src/antiocr/anti_ocr.py

```diff
@@ -0,0 +1,215 @@
+        Parameters
+        ----------
+        csv_path: str
+            Path to the eye-tracking CSV file.
+        page_name: str
+            Page/stimulus identifier to extract.
+        session: str
+            Recording-session label to extract.
+        output_path: str
+            Destination image file;
+            must end with ``.png``, ``.jpg``, ``.jpeg`` or ``.bmp``.
+
+        Raises
+        ------
+        ValueError
+            If the user cancels the column-mapping dialog.
+        """
+        mapping = self.mapping
+        if mapping is None:
+            root = tk.Tk()
+            root.withdraw()
+            mapping = ColumnMappingDialog(root, title='Column Mapping').result
+            root.destroy()
+
+            if mapping is None:
+                raise ValueError(
+                    'Column mapping configuration cancelled by user.',
+                )
+
+        column_mapping = {
+            mapping['pixel_x']: 'pixel_x',
+            mapping['pixel_y']: 'pixel_y',
+            mapping['interest_area_label']: 'word',
+            mapping['recording_session']: 'recording_session',
+            mapping['page_name']: 'page_name',
+        }
+
+        for filter_col in mapping['filter_columns']:
+            column_mapping[filter_col] = filter_col
+
+        try:
+            csv_file = Path(csv_path)
+
+            if not csv_file.is_file():
+                print(f"ERROR: CSV file not found: {csv_path}")
+                return
+
+            print(f"Loading gaze data from: {csv_file}")
+
+            df = pd.read_csv(
+                csv_file,
+                sep=None,
+                engine='python',
+                encoding='utf-8-sig',
+                usecols=list(column_mapping.keys()),
+            )
+            df.rename(columns=column_mapping, inplace=True)
+            df['normalized_page_name'] = df['page_name'].astype(
+                str,
+            ).apply(self._normalize_stimulus_name)
+
+            normalized_stimulus_name = self._normalize_stimulus_name(page_name)
+            filter_conditions = (
+                (df['recording_session'] == session) &
+                (df['normalized_page_name'] == normalized_stimulus_name)
+            )
+
+            if isinstance(mapping['filter_columns'], dict):
+                for col, allowed in mapping['filter_columns'].items():
+                    if col not in df.columns:
+                        print(
+                            f"WARNING: Filter column '{col}' not found; "
+                            f"ignoring filter.",
+                        )
+                        continue
+                    filter_conditions &= df[col].isin(allowed)
+            else:
+                print(
+                    "WARNING: 'filter_columns' is not a dictionary; "
+                    'skipping filters.',
+                )
+
+            df = df[filter_conditions].copy()
+
+        except (pd.errors.ParserError, FileNotFoundError) as e:
```

Comment:
try blocks should be kept minimal, you can move this huge try block into two, minimal for each exception that can happen